AutoMatrix/AutoMatrix.py

```diff
@@ -468,7 +468,7 @@ def ProcessVolume(self)-> None:
                         print("not a .nii.gz")
 
                     self.UpdateTime()
-                    if extension_scan!=".nii.gz":
+                    if extension_scan!=".nii.gz" and extension_scan!=".nrrd":
                         model = slicer.util.loadModel(scan)
                     else :
                         model = slicer.util.loadVolume(scan)
```

Comment:
You could try something like this, which will enable _all_ Slicer volume node types:

```py
volume_storage = slicer.vtkMRMLVolumeStorageNode()
model_storage = slicer.vtkMRMLVolumeStorageNode()

if volume_storage.GetSupportedFileExtension(scan):
  model = slicer.util.loadVolume(scan)
elif model_storage.GetSupportedFileExtension(scan):
  model = slicer.util.loadModel(scan)
else:
  continue  # or raise, log, etc.
```

`GetSupportedFileExtension` takes a filename and extracts its extension if the storage node can handle it, or returns `''` if not supported.

You could also use `slicer.util.loadNodeFromFile(scan)`, but this would attempt to load _any_ file type, not restricted to just volumes or models. It probably isn't what you want.

Reply:
I don't have time right now to make the change but I'm gonna do it thanks !